### ISSUE TITLE
drop trailing operators when executing selection

### ISSUE THREAD

**Issue by kevinushey** [rstudio/rstudio#8589]
Authors of R code that build results in a "pipeline" style will often have code of the form:

```
data %>%
    operation1() %>%
    operation2() %>%
    operation3() %>%
    ...
```

When building such a pipeline, it's useful to be able to execute subsets of that pipeline. One can do that by manually highlighting + selecting the code explicitly to be run, but this is usually not that ergonomic (since the cursor needs to be moved backwards to exclude the trailing operator on the to-be-executed line).

As a feature, we could drop any trailing operators when executing a multi-line expression. (It's relatively unlikely that a user would explicitly select a multi-line expression that is incomplete and requires further input, so we could infer that the user intent is to execute the code without the trailing operator.)

**Comment by warnes** [comment]
One note of caution: 

I'm one of those people who often finds myself selecting part of a pipeline series to execute, and I sometimes actually use R's expectation that the presence of the trailing operator indicates the expression isn't complete so I can go down a few lines and run another section of the pipeline.

Thus, I think it would be wise to provide a mechanism to control this behavior.   

Perhaps have both *actions* available with one mapped to the standard shortcut (e.g. <Control><Enter>) and the other to a similar key sequence (e.g. <Alt><Enter>) with a simple way to swap them.

Alternatively, provide an action bar button to toggle the state.

**Comment by iago-pssjd** [comment]
I agree with @warnes. I understand and agree with the usefulness of the feature proposed, to me too, but it can lead many times to run the wrong expression (and sometimes, correct it can be expensive), while the current way gives the chance to do a more careful selection of the suitable lines (after force an error if necessary).

So, if this feature goes forward, I would expect at least it was configurable in RStudio, making one behaviour as default and the other optional.
